api/handler/trigger.go

```diff
@@ -145,6 +153,31 @@ func removeTrigger(writer http.ResponseWriter, request *http.Request) {
 	}
 }
 
+// TriggerDeletionMiddleware middleware .
+func triggerDeletionMiddleware() func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		fn := func(w http.ResponseWriter, r *http.Request) {
+			db := middleware.GetDatabase(r)
+
+			trigger, err := controller.GetTrigger(db, middleware.GetTriggerID(r))
+			if err == nil {
+				userLogin := middleware.GetLogin(r)
+				auth := middleware.GetAuth(r)
+
+				if auth.IsEnabled() && !auth.IsAdmin(userLogin) ||
```

Comment:
Чёт кошмарно длинное условие вышло, может стоит вытащиь проверкив переменные
```suggestion
				admin := auth.IsEnabled() && !auth.IsAdmin(userLogin) 
				owner := trigger.CreatedBy == userLogin && auth.CanDeleteTriggersRegexp.MatchString(trigger.CreatedBy)
				
				if admin || owner {
```

Reply:
давай